libvast/src/system/start_command.cpp

```diff
@@ -61,6 +61,9 @@ caf::message start_command_impl(start_command_extra_steps extra_steps,
   if (!parsers::endpoint(str, node_endpoint))
     return caf::make_message(
       make_error(ec::parse_error, "invalid endpoint", str));
+  // Default to port 42000/tcp if none is set.
+  if (node_endpoint.port.type() == port::unknown)
+    node_endpoint.port = port{defaults::system::endpoint_port, port::tcp};
```

Comment:
I'm not following the logic, maybe a comment is missing, but where is the jump from "unknown port type" to "let's use the defaults?

Reply:
I've reimplemented this independent of the internals of the endpoint parser by making the port in an endpoint optional.

Reply:
Yes, `optional` semantics make much more sense here.